src/App.tsx

```diff
@@ -1,14 +1,59 @@
-/* eslint-disable max-len */
-import React from 'react';
+// App.tsx
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<string>('all');
+  const [query, setQuery] = useState<string>('');
+  const [isTodoModalShown, setIsTodoModalShown] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(data => {
+        setTodos(data);
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  const filteredTodos = todos.filter(todo => {
+    if (filter === 'all') {
+      return todo.title.toLowerCase().includes(query.trim().toLowerCase());
+    } else if (filter === 'active') {
+      return (
+        !todo.completed &&
+        todo.title.toLowerCase().includes(query.trim().toLowerCase())
+      );
+    } else if (filter === 'completed') {
```

Comment:
it is good place to use switch...case instead if...else.
better to create enum for filter.